### DISCUSSION TITLE
Is there a way to add custom code into a callback

### DISCUSSION THREAD

**Discussion by interactivetech** [ultralytics/yolov5#8656] [Q&A]
Hi There,

Big fan of YoloV5 here! I am interested in reporting per class AP after every epoch, and I was wondering if there are code snippets in adding/registering additional code into certain callbacks.

I see in utils.callbacks, there is a function to register hooks, is this where I can create a function and have it execute when `on_fit_epoch_end` occurs? Do I need to manually add additional code into on_fit_epoch_end function? This is the functionality I am trying to achieve:
```
# 1. create custom function
def run():
  print("print AP...")

 callbacks = Callbacks()
# 2. register function to execute when on_fit_epoch_end occurs
callbacks.register_action(hook='on_fit_epoch_end',name='run',callback=run)
```

**Comment by glenn-jocher** [comment]
@interactivetech yes the above look correct for adding function to callbacks, but you can also just set `verbose=True` to output per-class mAP after every epoch:
https://github.com/ultralytics/yolov5/blob/92e47b85d952274480c8c5efa5900e686241a96b/val.py#L109
